src/connstat.cpp

```diff
@@ -0,0 +1,169 @@
+#include "connstat.h"
+
+#include <netlink/netlink.h>
+#include <netlink/socket.h>
+#include <netlink/msg.h>
+#include <netlink/route/link.h>
+#include <net/if.h>
+
+namespace dht {
+namespace net {
+
+/* forward declarations */
+static int nlcb(struct nl_msg*, void*);
+
+ConnectivityStatus::ConnectivityStatus()
+    : nlsk(nlsk_init())
+    , logger_(new dht::Logger)
+{
+    if (not nlsk_setup(nlsk.get()))
+        throw std::runtime_error("nlsk_setup() failed\n");
+
+    thrd_ = std::thread([this] () { nl_event_loop_thrd(nlsk.get()); });
+    thrd_.detach();
```

Comment:
never detach thread, especially since you capture this (would cause use-after-free).

Reply:
without it i get:
_terminate called without an active exception_